src/device/manager/device_discovery.rs

```diff
@@ -0,0 +1,225 @@
+                lines.len(),
+                lines
+            );
+            return None;
+        }
+
+        let device_name = lines[0].trim().to_string();
+        let manufacturer = lines[1].trim().to_string();
+
+        let mac_address_line = lines[2].trim();
+        let ip_address_line = lines[3].trim();
+
+        if !mac_address_line.starts_with("MAC Address:- ")
+            || !ip_address_line.starts_with("IP Address:- ")
+        {
+            warn!("auto_create: network: Unexpected format in MAC Address or IP Address lines.");
+            return None;
+        }
+
+        let mac_address = mac_address_line
+            .trim_start_matches("MAC Address:- ")
+            .to_string();
+        let ip_address_str = ip_address_line.trim_start_matches("IP Address:- ").trim();
+
+        let cleaned_ip_address_str = ip_address_str
+            .split('.')
+            .map(|octet| {
+                let trimmed = octet.trim_start_matches('0');
+                if trimmed.is_empty() {
+                    "0"
+                } else {
+                    trimmed
+                }
+            })
+            .collect::<Vec<&str>>()
+            .join(".");
+
+        let ip_address = match cleaned_ip_address_str.parse::<Ipv4Addr>() {
+            Ok(ip) => ip,
+            Err(err) => {
+                warn!(
+                    "auto_create: network: Failed to parse IP address: {cleaned_ip_address_str}, details: {err}"
+                );
+                return None;
+            }
+        };
+
+        Some(DiscoveryResponse {
+            device_name,
+            manufacturer,
+            mac_address,
+            ip_address,
+        })
+    }
+}
+
+pub fn network_discovery() -> Option<Vec<SourceSelection>> {
+    // Create a UDP socket
+    let socket = std::net::UdpSocket::bind("0.0.0.0:0").unwrap(); // Bind to any available port
+    socket.set_broadcast(true).unwrap(); // Enable broadcast
+
+    let broadcast_addr = "255.255.255.255:30303";
+    let discovery_message = "Discovery";
+
+    socket
+        .send_to(discovery_message.as_bytes(), broadcast_addr)
+        .unwrap();
+
+    socket
+        .set_read_timeout(Some(std::time::Duration::from_secs(2)))
+        .unwrap();
+
+    let mut buf = [0; 1024];
+    let mut responses = Vec::new();
+
+    loop {
+        match socket.recv_from(&mut buf) {
+            Ok((size, src)) => {
+                let response = std::str::from_utf8(&buf[..size]).unwrap_or("[Invalid UTF-8]");
+
+                if let Some(discovery_response) = DiscoveryResponse::from_response(response) {
+                    responses.push(discovery_response);
+                } else {
+                    warn!(
+                        "auto_create: network: Failed to parse the discovery response from: {src}"
+                    );
+                }
+            }
+            Err(e) => {
+                warn!("auto_create: network: Timeout or error receiving response: {e}");
```

Comment:
watch out for your consistency:
```suggestion
            Err(err) => {
                warn!("auto_create: network: Timeout or error receiving response: {err}");
```